### PR TITLE
gravityCompensator: fix crash due to a move operator after a resize

### PR DESCRIPTION
**Pull request by Nicogene**
It needs more investigations, but for now this make works gravityCompensator

TO BE TESTED on the robot (cc @violadelbono @julijenv )

### CODE REVIEW

**Comment by Nicogene** [comment]
Sorry wrong base branch, closing and reopening with the right base branch

**Review by pattacini — CHANGES_REQUESTED** [review]
Thanks @Nicogene for the fix 👍🏻 

Could you please polish the changes in order to keep only the relevant fix and remove the formatting?

**Comment by traversaro** [comment]
I would avoid to name a branch `v1.17.1`, as that is the style that we use for tags.

**Comment by pattacini** [comment]
>I would avoid to name a branch v1.17.1, as that is the style that we use for tags.

The branch is just temporary as we're gonna churn out a tag `v1.17.1` and soon afterward rebase `master` and `devel` on top.

**Comment by pattacini** [comment]
Uhm maybe there might be a conflict between the branch and the tag as their name is the same (and conceptually they are so pointers). Let me rename the branch then.

**Comment by traversaro** [comment]
> 
> 
> > I would avoid to name a branch v1.17.1, as that is the style that we use for tags.
> 
> The branch is just temporary as we're gonna churn out a tag `v1.17.1` and soon afterward rebase `master` and `devel` on top.

Ack, to avoid confusion for the future we can use a different name for the temporary branch, something like `b1.17.1` . Anyhow it is noto really important.

**Comment by pattacini** [comment]
Branch name changed into `robotology:fix/gravitycompensator` and PR updated.

File: src/libraries/iKin/src/iKinFwd.cpp [hunk 1]
```diff
@@ -2711,8 +2711,8 @@ void iCubInertialSensor::allocate(const string &_type)
         setHN(getHN()*HN);
     }
 
-    // further displacement for v2.6
-    if (getType()=="v2.6")
+    // further displacement for > 2.6
+    if (version >= 2.6)
```
Comment (pattacini) [inline src/libraries/iKin/src/iKinFwd.cpp:2715]:
```suggestion
    if (version>=2.6)
```

just for consistency in formatting

File: src/libraries/iKin/src/iKinFwd.cpp [hunk 2]
```diff
@@ -2711,8 +2711,8 @@ void iCubInertialSensor::allocate(const string &_type)
         setHN(getHN()*HN);
     }
 
-    // further displacement for v2.6
-    if (getType()=="v2.6")
+    // further displacement for > 2.6
```
Comment (pattacini) [inline src/libraries/iKin/src/iKinFwd.cpp:2714]:
```suggestion
    // further displacement for >= 2.6
```

**Review by pattacini — APPROVED** [review]
Actually, we don't really need to perform careful tests.

**Pull request merged**